timesketch/frontend-ng/src/components/Explore/EventList.vue

```diff
@@ -0,0 +1,946 @@
+    itemsPerPage: {
+      type: Number,
+      default: 40,
+    },
+  },
+  data() {
+    return {
+      columnHeaders: [
+        {
+          text: '',
+          value: 'field',
+        },
+      ],
+      tableOptions: {
+        itemsPerPage: this.itemsPerPage,
+      },
+      currentItemsPerPage: this.itemsPerPage,
+      expandedRows: [],
+      selectedFields: [{ field: 'message', type: 'text' }],
+      searchColumns: '',
+      columnDialog: false,
+      saveSearchMenu: false,
+      saveSearchFormName: '',
+      selectedEventTags: [],
+      tagConfig: {
+        good: { color: 'green', textColor: 'white', label: 'mdi-check-circle-outline' },
+        bad: { color: 'red', textColor: 'white', label: 'mdi-alert-circle-outline' },
+        suspicious: { color: 'orange', textColor: 'white', label: 'mdi-help-circle-outline' },
+      },
+      searchInProgress: false,
+      currentPage: 1,
+      eventList: {
+        meta: {},
+        objects: [],
+      },
+      currentQueryString: '',
+      currentQueryFilter: defaultQueryFilter(),
+      selectedEvents: [],
+      displayOptions: {
+        isCompact: false,
+        showTags: true,
+        showEmojis: true,
+        showMillis: false,
+        showTimelineName: true,
+      },
+      showHistogram: false,
+      branchParent: null,
+    }
+  },
+  computed: {
+    sketch() {
+      return this.$store.state.sketch
+    },
+    meta() {
+      return this.$store.state.meta
+    },
+    totalHits() {
+      if (this.eventList.meta.es_total_count > 0 && this.eventList.meta.es_total_count_complete === 0) {
+        return this.eventList.meta.es_total_count
+      }
+      return this.eventList.meta.es_total_count_complete || 0
+    },
+    totalHitsForPagination() {
+      // Opensearch has a limit of 10k events when paginating
+      return this.eventList.meta.es_total_count || 0
+    },
+    totalTime() {
+      return this.eventList.meta.es_time / 1000 || 0
+    },
+    fromEvent() {
+      return this.currentQueryFilter.from || 1
+    },
+    toEvent() {
+      if (this.totalHits < this.currentQueryFilter.size) {
+        return
+      }
+      return parseInt(this.currentQueryFilter.from) + parseInt(this.currentQueryFilter.size)
+    },
+    timeFilterChips: function () {
+      return this.currentQueryFilter.chips.filter((chip) => chip.type.startsWith('datetime'))
+    },
+    currentSearchNode() {
+      return this.$store.state.currentSearchNode
+    },
+    headers() {
+      let baseHeaders = [
+        {
+          text: '',
+          value: 'data-table-select',
+        },
+        {
+          value: 'actions',
+          width: '105',
+        },
+        {
+          text: 'Datetime (UTC)',
+          align: 'start',
+          value: '_source.timestamp',
+          width: '200',
+        },
+        {
+          value: '_source.comment',
+          width: '40',
+        },
+      ]
+      let extraHeaders = []
+      this.selectedFields.forEach((field) => {
```

Comment:
The changes introduced in the [last commit](https://github.com/google/timesketch/pull/2592/commits/4efc22847b2f54786dc731e59565917799cbe725) result in `this.selectedFields` being undefined at this point. 
This results in a broken search and TS throwing the following error:
```
Error in render: "TypeError: Cannot read properties of undefined (reading 'forEach')"
found in

---> <TsEventList> at src/components/Explore/EventList.vue
       <VCard>
         <Explore> at src/views/Explore.vue
           <Canvas> at src/views/Canvas.vue
             <Sketch> at src/views/Sketch.vue
               <VMain>
                 <VApp>
                   <App> at src/App.vue
                     <Root>
```
and
```
 TypeError: Cannot read properties of undefined (reading 'forEach')
    at VueComponent.headers (EventList.vue?1150:519:1)
```